### ISSUE TITLE
Refactor to use `lightning.LightningDataModule`

### ISSUE THREAD

**Issue by charlesfrye** [transmissions11/bistro#4]
After #3 to get the model code organized, use a `LightningDataModule` to get the data code organized. Again, the [docs](https://lightning.ai/docs/pytorch/stable/data/datamodule.html#lightningdatamodule) are helpful. In particular, there's a section that notes [how to use it without the `Trainer`](https://lightning.ai/docs/pytorch/stable/data/datamodule.html#datamodules-without-lightning).

This is where I'd put the `tokenizer`, actually, along with some of the masking logic -- unless you move the masking logic into a custom loss (with `CrossEntropyLoss(...reduce='none')` followed by your own code).
